src/promptflow/tests/executor/conftest.py

```diff
@@ -0,0 +1,115 @@
+import multiprocessing
+from asyncio import Queue
+from contextlib import contextmanager
+from functools import wraps
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+from pytest_mock import MockerFixture
+from sdk_cli_test.recording_utilities import RecordStorage, mock_tool, recording_array_extend, recording_array_reset
+
+from promptflow.executor._line_execution_process_pool import _process_wrapper
+
+PROMPTFLOW_ROOT = Path(__file__) / "../../.."
+RECORDINGS_TEST_CONFIGS_ROOT = Path(PROMPTFLOW_ROOT / "tests/test_configs/node_recordings").resolve()
+
+
+@contextmanager
+def apply_recording_injection_if_enabled():
+    # multiprocessing.get_context("spawn").Process = MockSpawnProcess
+    # multiprocessing.Process = MockSpawnProcess
+
+    patches = []
+    if RecordStorage.is_replaying_mode() or RecordStorage.is_recording_mode():
+        file_path = RECORDINGS_TEST_CONFIGS_ROOT / "node_cache.shelve"
+        RecordStorage.get_instance(file_path)
+
+        from promptflow._core.tool import tool as original_tool
+
+        mocked_tool = mock_tool(original_tool)
+        patch_targets = ["promptflow._core.tool.tool", "promptflow._internal.tool", "promptflow.tool"]
+
+        for target in patch_targets:
+            patcher = patch(target, mocked_tool)
+            patches.append(patcher)
+            patcher.start()
+
+    try:
+        yield
+    finally:
+        for patcher in patches:
+            patcher.stop()
+
+
+def recording_injection_decorator(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        multiprocessing.get_context("spawn").Process = MockSpawnProcess
+        multiprocessing.Process = MockSpawnProcess
+        with apply_recording_injection_if_enabled():
+            return func(*args, **kwargs)
+
+    return wrapper
+
+
+def recording_injection_decorator_compatible_with_spawn(mock_class):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            original_process_class = multiprocessing.get_context("spawn").Process
+            multiprocessing.get_context("spawn").Process = mock_class
+            multiprocessing.Process = mock_class
```

Comment:
Seems need to change name of `mock_class` to `MockProcessClass`

Reply:
this is to make this method generic